addons/safemode/functions/fn_safeWeapon.sqf

```diff
@@ -0,0 +1,25 @@
+/*
+ * Author: Brostrom.A
+ * Safe the given weapon if unlocked.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Weapon <STRING>
+ * 2: Muzzle <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [ACE_player, currentWeapon ACE_player, currentMuzzle ACE_player] call ace_safemode_fnc_safeWeapon
+ *
+ * Public: yes
+ */
+
+params ["_unit", "_weapon", "_muzzle"];
+
+private _safedWeapons = _unit getVariable [QGVAR(safedWeapons), []];
+
+if !(_weapon in _safedWeapons) then { 
```

Comment:
Public functions should not be case sensitive.

Reply:
is there a quick way to convert a array to lower case?

Reply:
`if ((_safedWeapons findIf {_weapon == _x}) == -1) then {`
should do it

Reply:
The script uses config case. So you can just convert random case to config case:
```sqf
_weapon = configName (configFile >> "CfgWeapons" >> _weapon);
```